node/rpc/src/lib.rs

```diff
@@ -429,130 +426,65 @@ async fn handle_connection_and_log<KS: KeyStore + Send + Sync + 'static>(
-                                            },
-                                        );
-                                    })
-                                    .await
-                                    .unwrap_or_else(|_| {
-                                        error!("error sending on socket {:?}", addr);
-                                        None
-                                    });
-
-                                    // wait for join handle to complete if there is error and send it over the network and cancel streaming
-                                    let error_join_send = ws_sender_clone.clone();
-                                    let handle_events_out = events_out_clone.clone();
-                                    task::spawn(async move {
-                                        if let Some(handle) = join_handle {
-                                            handle
-                                                .map_err(|e| async move {
-                                                    send_error(
-                                                        3,
-                                                        &error_join_send,
-                                                        format!(
-                                                            "channel id {:}, error {:?}",
-                                                            chain_notify_count_curr,
-                                                            e.message()
-                                                        ),
-                                                    )
-                                                    .await
-                                                    .unwrap_or_else(|e| {
-                                                        error!(
-                                                            "error {:?} on socket {:?}",
-                                                            e.message(),
-                                                            addr
-                                                        )
-                                                    });
-                                                })
-                                                .await
-                                                .unwrap_or_else(|_| {
-                                                    error!("error sending on socket {:?}", addr)
-                                                });
+                                        let mut head_changes = cs.sub_head_changes();
 
-                                            handle_events_out
-                                                .write()
+                                        // First response should be the count serialized.
+                                        // This is based on internal golang channel rpc handling
+                                        // needed to match Lotus.
```

Comment:
This doesnt return a JSON response with the channel ID. You need to do something like this:
```                                        
                                        let data = SubscribeChannelIDResponse {
                                            json_rpc: "2.0",
                                            result: chain_notify_count_curr,
                                            id: call.id.unwrap().unwrap(),
                                        };
                                        if let Err(e) =
                                            send_response(&ws_sender, &data)
                                                .await
                                        {
                                            let msg = e.message();
                                            send_error(3, &ws_sender, msg).await;
                                            return;
                                        }
```

Reply:
Ah my mistake, yeah it sends just the response as json, and not wrapped in the RPC response, fixing now. I forgot that was implicitly being done when routed through the rpc table